deptofdefense/move.mil
# Use data from free_zipcode_data to power location search for ZIP codes

## Checklist

I have…


## Summary of Changes

This pull request utilizes data from the GitHub project `free_zipcode_data` to power the ZIP code search functionality in the locator maps instead of relying on Google for it.

In the unlikely event of the user appending `?query=62225` (as an example) to the locator map URL, or following a link to the page with that in the URL, the results will now say "Displaying results near Scott Air Force Base, IL 62225" (the city and state associated with the ZIP code) instead of "Displaying results near 38.5426987, -89.8551818" (the lat / lng associated with the ZIP code).

That said, if the user follows the normal route of typing "62225" in the search box, the page will still show the coordinates instead of the city state. Addressing _that_ will require a more extensive change to the `locations_controller.rb`.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. git checkout `my_zipcode_gem`,
1. set up development dependencies according to [CONTRIBUTING.md](https://github.com/deptofdefense/move.mil/blob/master/CONTRIBUTING.md),
1. run `bin/rails server`, and
1. load up [http://localhost:3000](http://localhost:3000) in the Web browser of your choice.

## Screenshots

<details><summary>Results from searching for 62225 - same as before</summary>
<img width="1118" alt="screen shot 2018-03-22 at 12 18 27 pm" src="https://user-images.githubusercontent.com/26554826/37783180-377fc8de-2dcb-11e8-9396-e1526fe67019.png">

</details>

<details><summary>Results with ?query=62225 in the URL</summary>
<img width="1119" alt="screen shot 2018-03-22 at 12 18 02 pm" src="https://user-images.githubusercontent.com/26554826/37783199-44a61022-2dcb-11e8-9f4a-e7c7fa0b6745.png">

</details>
